### PR TITLE
Add cn docs for Transformer's APIs.

### PR DESCRIPTION
**Pull request by xiemoyuan**
Add cn docs for Transformer's APIs, include `MultiHeadAttention`, `TransformerEncoderLayer`, `TransformerEncoder`, `TransformerDecoderLayer`, `TransformerDecoder`, `Transformer`.

The cn docs pre-view links as follow：

`MultiHeadAttention`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.html
MultiHeadAttention

`TransformerEncoderLayer`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.html
TransformerEncoderLayer1
TransformerEncoderLayer2

`TransformerEncoder`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/TransformerEncoder_cn.html
TransformerEncoder

`TransformerDecoderLayer`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.html
TransformerDecoderLayer1
TransformerDecoderLayer2

`TransformerDecoder`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/TransformerDecoder_cn.html
TransformerDecoder

`Transformer`: http://gzhxy-inf-bce36a39de.gzhxy.baidu.com:8121/documentation/docs/zh/api/paddle/nn/layer/transformer/Transformer_cn.html
Transformer1
Transformer2

### CODE REVIEW

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+MultiHeadAttention
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst:3]: 前面需要加个标签，方便其他文件引用，格式为
【.. _cn_path_to_api:】然后再空一行 写正文，可以参考 [paddle.add](https://raw.githubusercontent.com/PaddlePaddle/FluidDoc/develop/doc/paddle/api/paddle/tensor/math/add_cn.rst)

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
+TransformerDecoderLayer
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst:3]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.MultiHeadAttention(embed_dim, num_heads, dropout=0.0, kdim=None, vdim=None, need_weights=False, weight_attr=None, bias_attr=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst]: `paddle.nn.MultiHeadAttention`，写推荐别名~

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+.. code-block:: python
+
+   import paddle
+
+   # encoder input: [batch_size, sequence_length, d_model]
+   query = paddle.rand((2, 4, 128))
+   # self-attention mask: [batch_size, num_heads, query_len, query_len]
+   attn_mask = paddle.rand((2, 2, 4, 4))
+   multi_head_attn = paddle.MultiHeadAttention(128, 2)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst]: `paddle.nn.MultiHeadAttention` 吧？

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.TransformerDecoderLayer(d_model, nhead, dim_feedforward, dropout=0.1, activation='relu', attn_dropout=None, act_dropout=None, normalize_before=False, weight_attr=None, bias_attr=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
+TransformerDecoder
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst:3]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.TransformerDecoder(decoder_layer, num_layers, norm=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst [hunk 8]
```diff
@@ -0,0 +1,39 @@
+TransformerEncoderLayer
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst:3]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.TransformerEncoderLayer(d_model, nhead, dim_feedforward, dropout=0.1, activation='relu', attn_dropout=None, act_dropout=None, normalize_before=False, weight_attr=None, bias_attr=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst [hunk 10]
```diff
@@ -0,0 +1,33 @@
+TransformerEncoder
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst:3]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.TransformerEncoder(encoder_layer, num_layers, norm=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst [hunk 12]
```diff
@@ -0,0 +1,57 @@
+Transformer
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst:3]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+-------------------------------
+
+.. py:class:: paddle.nn.layer.transformer.Transformer(d_model=512, nhead=8, num_encoder_layers=6, num_decoder_layers=6, dim_feedforward=2048, dropout=0.1, activation='relu', attn_dropout=None, act_dropout=None, normalize_before=False, weight_attr=None, bias_attr=None, custom_encoder=None, custom_decoder=None)
```
Comment (TCChenlong) [inline doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst]: 同上

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst (hunk 1 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst (hunk 2 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst (hunk 3 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst (hunk 4 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/MultiHeadAttention_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst (hunk 5 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoderLayer_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst (hunk 6 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst (hunk 7 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerDecoder_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst (hunk 8 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst (hunk 9 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoderLayer_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst (hunk 10 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst (hunk 11 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/TransformerEncoder_cn.rst]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst (hunk 12 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst:3]: Done. Thanks!

File: doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst (hunk 13 above)
Comment (xiemoyuan) [reply doc/paddle/api/paddle/nn/layer/transformer/Transformer_cn.rst]: Done. Thanks!

**Review by TCChenlong — APPROVED** [review]
LGTM

**Pull request merged**